vectordraw.py

```diff
@@ -48,7 +70,41 @@ def check_tip_x(check, vectors):
 def check_tip_y(check, vectors):
     vec = vectors[check['vector']]
     if _check_coordinate(check, vec.tip.y):
-        return 'Vector {} does not tip at correct point.'.format(vec.name)
+        return 'Vector {} does not end at correct point.'.format(vec.name)
+
+def _check_coords(vec, expected, tolerance):
+    for expected_coords, vec_coords in ([expected[0], vec.tail], [expected[1], vec.tip]):
+        if expected_coords[0] == '_':
+            if expected_coords[1] != '_':
+                # Only y given; not interested in x.
+                if abs(expected_coords[1] - vec_coords.y) > tolerance:
+                    return True
+        elif expected_coords[1] == '_':
+            # Only x given; not interested in y.
+            if abs(expected_coords[0] - vec_coords.x) > tolerance:
+                return True
+        else:
+            # Both coordinates given; check distance from expected point.
+            x = expected_coords[0] - vec_coords.x
+            y = expected_coords[1] - vec_coords.y
+            if math.hypot(x, y) > tolerance:
+                return True
+    return False
+
+def check_coords(check, vectors):
+    vec = vectors[check['vector']]
+    expected = check['expected']
+    tolerance = check.get('tolerance', 1.0)
+    if _check_coords(vec, expected, tolerance):
+        return 'Vector {} coordinates are not correct.'.format(vec.name)
+
+def check_segment_coords(check, vectors):
+    vec = vectors[check['vector']]
+    expected = check['expected']
+    tolerance = check.get('tolerance', 1.0)
+    if _check_coords(vec, expected, tolerance) and \
+            _check_coords(vec.opposite(), expected, tolerance):
+        return 'Segment {} coordinates are not correct.'.format(vec.name)
```

Comment:
An alternative to checking the vector and its opposite is to normalise the direction of segments – both the expected values as well as the vectors entered by the student.  An easy way to do so would be to lexicographically sort the points.
Your approach is completely fine, no need to change anything.  You might have gotten away with slightly less work by subclassing `Vector` with a `Segment` class that normalises the point order, but you definitely won't save any work by doing it now. :)